lib/extensions/ar_virtual.rb

```diff
@@ -617,10 +617,11 @@ def find_with_associations
       def select(*fields)
         return super if block_given? || fields.empty?
         # support virtual attributes by adding an alias to the sql phrase for the column
+        # it does not add an as() if the column already has an as
         # this code is based upon _select()
         fields.flatten!
         fields.map! do |field|
-          if virtual_attribute?(field) && (arel = klass.arel_attribute(field))
+          if virtual_attribute?(field) && (arel = klass.arel_attribute(field)) && arel.respond_to?(:as)
```

Comment:
Curious if there would be any performance implications to adding these extra checks to every query (not just from this PR, but prior ones as well), but it is probably negligible and not worth investigating.  Just making a note of it.